rocketmq-spring-boot/src/main/java/org/apache/rocketmq/spring/core/RocketMQTemplate.java

```diff
@@ -76,6 +80,179 @@ public void setMessageQueueSelector(MessageQueueSelector messageQueueSelector) {
+        requestAsync(destination, message, requestCallback, producer.getSendMsgTimeout(), delayLevel);
+    }
+
+    public void requestAsyncOrderly(String destination, Message<?> message, RequestCallback requestCallback, String hashKey) {
+        requestAsyncOrderly(destination, message, requestCallback, hashKey, producer.getSendMsgTimeout());
+    }
+
+    public void requestAsyncOrderly(String destination, Object payload, RequestCallback requestCallback, String hashKey) {
+        requestAsyncOrderly(destination, payload, requestCallback, hashKey, producer.getSendMsgTimeout());
+    }
+
+    public void requestAsyncOrderly(String destination, Message<?> message, RequestCallback requestCallback, String hashKey, long timeout) {
+        requestAsyncOrderly(destination, message, requestCallback, hashKey, timeout, 0);
+    }
+
+    public void requestAsyncOrderly(String destination, Object payload, RequestCallback requestCallback, String hashKey, long timeout) {
+        requestAsyncOrderly(destination, payload, requestCallback, hashKey, timeout, 0);
+    }
+
+    public void requestAsyncOrderly(String destination, Message<?> message, RequestCallback requestCallback, String hashKey, long timeout, int delayLevel) {
+        if (Objects.isNull(message) || Objects.isNull(message.getPayload())) {
+            log.error("send request message failed. destination:{}, message is null ", destination);
+            throw new IllegalArgumentException("`message` and `message.payload` cannot be null");
+        }
+
+        try {
+            org.apache.rocketmq.common.message.Message rocketMsg = RocketMQUtil.convertToRocketMessage(rocketMQMessageConverter.getMessageConverter(),
+                charset, destination, message);
+            if (delayLevel > 0) {
+                rocketMsg.setDelayTimeLevel(delayLevel);
+            }
+            if (Objects.isNull(hashKey) || hashKey.isEmpty()) {
+                producer.request(rocketMsg, requestCallback, timeout);
+            } else {
+                producer.request(rocketMsg, messageQueueSelector, hashKey, requestCallback, timeout);
+            }
+        } catch (Exception e) {
+            log.error("send request message failed. destination:{}, message:{} ", destination, message);
+            throw new MessagingException(e.getMessage(), e);
+        }
+    }
+
+    public void requestAsyncOrderly(String destination, Object payload, RequestCallback requestCallback, String hashKey, long timeout, int delayLevel) {
+        Message<?> message = MessageBuilder.withPayload(payload).build();
+        requestAsyncOrderly(destination, message, requestCallback, hashKey, producer.getSendMsgTimeout(), delayLevel);
+    }
+
     /**
      * <p> Send message in synchronous mode. This method returns only when the sending procedure totally completes.
      * Reliable synchronous transmission is used in extensive scenes, such as important notification messages, SMS
```

Comment:
[Discuss] Sending spring message but returning rocketmq message may not be friendly for users, especially the full name of rocketmq message is so long. Do we need to expose rocketmq message to users？

Reply:
IMO, there are too many method names. Keep the same name, such as 'sendAndReceive ' and use different parameters to distinguish different types of methods.

Reply:
> [Discuss] Sending spring message but returning rocketmq message may not be friendly for users, especially the full name of rocketmq message is so long. Do we need to expose rocketmq message to users？

Yeah, I am not sure about that too, but I think users MAY need the attributes in RocketMQ Message, such as topic/properties.

> IMO, there are too many method names. Keep the same name, such as 'sendAndReceive ' and use different parameters to distinguish different types of methods.

I have thought about that in the first place. But I changed my mind because the current implementation will make the code simple and more friendly to users.
(Seems `sendXXOrdery` method can be merged sendXXX, but can not merge `requestSync` and `requestAsync` because of different return types the sync mode and async mode. )